src/http/api/storage.ts

```diff
@@ -0,0 +1,105 @@
+import type { Account } from 'account/account.entity';
+import type { AccountService } from 'account/account.service';
+import { type Result, error, getError, isError, ok } from 'core/result';
+import type { Context } from 'hono';
+import type { GCPStorageService } from 'storage/gcloud-storage/gcp-storage.service';
+import { v4 as uuidv4 } from 'uuid';
+
+const ALLOWED_IMAGE_TYPES = [
+    'image/jpg',
+    'image/jpeg',
+    'image/png',
+    'image/webp',
+];
+
+type FileValidationError = 'file-too-large' | 'file-type-not-supported';
+
+export function createStorageHandler(
+    accountService: AccountService,
+    storageService: GCPStorageService,
+) {
+    /**
+     * Handle an upload to GCloud Storage bucket
+     */
+    return async function handleUpload(ctx: Context) {
+        const logger = ctx.get('logger');
+        const bucket = storageService.getBucket();
+
+        const formData = await ctx.req.formData();
+        const file = formData.get('file') as File;
```

Comment:
I think this is a dangerous assertion - we don't know that `file` is the type of `File`! So the rest of our code becomes unsafe - we should check that is is a `File` with an instanceof check I think